### PR TITLE
Switch from Mapbox to PMTiles

### PR DESCRIPTION
**Pull request by brandonfcohen1**
Check it out here: https://vacant-lots-proj-f9q57ld18-clean-and-green-philly.vercel.app/

Instead of using mapbox for tile generation and hosting, I updated the script to create [pmtiles](https://docs.protomaps.com/pmtiles/) locally using tippecanoe and hosting on Google Cloud.

This allows for properties to be shown as points at low zoom levels and polygons at high zoom levels.

As a result, I had to move the map provider from mapbox to maplibre. Some functionality had to be rewritten.

Please pressure test the hell out of the map!

### CODE REVIEW

File: src/components/PropertyMap.tsx [hunk 1]
```diff
@@ -153,11 +206,23 @@ const PropertyMap: FC<PropertyMapProps> = ({
...
     let bbox: [PointLike, PointLike] | undefined = undefined;
-    const features = map.queryRenderedFeatures(bbox, {
-      layers: ["vacant_properties_tiles"],
-    });
 
-    setFeatureCount(features.length);
+    const features = map.queryRenderedFeatures(bbox, { layers });
+
+    //Get count of features if they are clustered
+    const cluteredFeatureCount = features.reduce(
```
Comment (paulhchoi) [inline src/components/PropertyMap.tsx:213]: typo here

File: src/components/PropertyMap.tsx [hunk 2]
```diff
@@ -325,27 +310,27 @@ const PropertyMap: FC<PropertyMapProps> = ({
     <div className="customized-map relative h-full w-full">
       <Map
         mapboxAccessToken={mapboxAccessToken}
```
Comment (paulhchoi) [inline src/components/PropertyMap.tsx:312]: can probably take this out now?

File: src/components/PropertyMap.tsx [hunk 3]
```diff
@@ -325,27 +310,27 @@ const PropertyMap: FC<PropertyMapProps> = ({
...
+        mapLib={maplibregl as any}
         initialViewState={{
           longitude: -75.15975924194129,
           latitude: 39.9910071520824,
           zoom,
         }}
-        mapStyle="mapbox://styles/mapbox/light-v10"
-        onMouseEnter={e => changeCursor(e, "pointer")}
-        onMouseLeave={e => changeCursor(e, "default")}
+        mapStyle="https://api.maptiler.com/maps/dataviz/style.json?key=dIagszPpXO3RgO1NNgzm"
```
Comment (paulhchoi) [inline src/components/PropertyMap.tsx:319]: should probably `env` this key?

**Review by paulhchoi — COMMENTED** [review]
small nits — @brandonfcohen1 this is amazing stuff, great work!

File: src/components/PropertyMap.tsx (hunk 1 above)
Comment (brandonfcohen1) [reply src/components/PropertyMap.tsx]: thx, fixed

File: src/components/PropertyMap.tsx (hunk 2 above)
Comment (brandonfcohen1) [reply src/components/PropertyMap.tsx:312]: yep, done

**Review by thansidwell — APPROVED** [review]
It looks good to me. There are things I'd consider tweaking but we can do that later.

**Pull request merged**